tests/core/test_wmi.py

```diff
@@ -357,12 +379,131 @@ def test_wql_filtering(self):
 
         # Check `_format_filter` logic
         no_filters = []
-        filters = [{'Name': "SomeName"}, {'Id': "SomeId"}]
+        filters = [{'Name': "SomeName", 'Id': "SomeId"}]
 
         self.assertEquals("", format_filter(no_filters))
-        self.assertEquals(" WHERE Id = 'SomeId' AND Name = 'SomeName'",
+        self.assertEquals(" WHERE ( Name = 'SomeName' AND Id = 'SomeId' )",
                           format_filter(filters))
 
+    def test_wql_multiquery_filtering(self):
+        """
+        Format the filters with multiple properties per instance to a comprehensive WQL `WHERE` clause.
+        """
+        from checks.libs.wmi import sampler
+        format_filter = sampler.WMISampler._format_filter
+
+        # Check `_format_filter` logic
+        no_filters = []
+        filters = [{'Name': "SomeName", 'Property1': "foo"}, {'Name': "OtherName", 'Property1': "bar"}]
+
+        self.assertEquals("", format_filter(no_filters))
+        self.assertEquals(" WHERE ( Property1 = 'bar' AND Name = 'OtherName' ) OR"
+                          " ( Property1 = 'foo' AND Name = 'SomeName' )",
+                          format_filter(filters))
+
+    def test_wql_empty_list(self):
+        """
+        Format filters to a comprehensive WQL `WHERE` clause skipping empty lists.
+        """
+
+        from checks.libs.wmi import sampler
+        format_filter = sampler.WMISampler._format_filter
+
+        filters = []
+        query = {}
+        query['User'] = ('=', 'luser')
+        query['SourceName'] = ('=', 'MSSQL')
+        query['EventCode'] = []
+        query['SomethingEmpty'] = []
+        query['MoreNothing'] = []
+
+        filters.append(query)
+
+        self.assertEquals(" WHERE ( SourceName = 'MSSQL' AND User = 'luser' )",
+                          format_filter(filters))
+
+    def test_wql_filtering_op_adv(self):
+        """
+        Format the filters to a comprehensive WQL `WHERE` clause w/ mixed filter containing regular and operator modified properties.
+        """
+        from checks.libs.wmi import sampler
+        format_filter = sampler.WMISampler._format_filter
+
+        # Check `_format_filter` logic
+        filters = [{'Name': "Foo%"}, {'Name': "Bar*", 'Id': ('>=', "SomeId")}, {'Name': "Zulu"}]
```

Comment:
Actually the wildcard was also implemented in the `event_log` check, but w/o the weird `*`-> `%` replacement, so I think we have another argument in favor of only allowing the `%` wildcard in filters :)